src/tasks/withdraw/value.ts

```diff
@@ -0,0 +1,115 @@
+import axios from "axios";
+import { BigNumber, constants } from "ethers";
+
+import { OrderKind } from "../../ts";
+import { TokenDetails } from "../ts/erc20";
+
+interface ApiTradeQuery {
+  network: string;
+  baseToken: string;
+  quoteToken: string;
+  kind: OrderKind;
+  amount: BigNumber;
+}
+
+interface PricedToken extends TokenDetails {
+  // Overrides existing field in TokenDetails. The number of decimals must be
+  // known to estimate the price
+  decimals: number;
+  // Amount of DAI wei equivalent to one unit of this token (10**decimals)
+  usdValue: BigNumber;
+}
+
+const REFERENCE_TOKEN: Record<
+  string,
+  { symbol: string; decimals: number; address: string }
+> = {
+  rinkeby: {
+    symbol: "DAI",
+    decimals: 18,
+    address: "0x5592ec0cfb4dbc12d3ab100b257153436a1f0fea",
+  },
+  mainnet: {
+    symbol: "DAI",
+    decimals: 18,
+    address: "0x6b175474e89094c44da98b954eedeac495271d0f",
+  },
+  xdai: {
+    // todo: replace with XDAI when native token price queries will be supported
+    // by the services.
+    symbol: "WXDAI",
+    decimals: 18,
+    address: "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d",
+  },
+} as const;
+
+function apiPriceUrl({
+  network,
+  baseToken,
+  quoteToken,
+  kind,
+  amount,
+}: ApiTradeQuery) {
+  // Using dev endpoint to avoid stressing the prod server with too many queries
+  return `https://protocol-${network}.dev.gnosisdev.com/api/v1/markets/${baseToken}-${quoteToken}/${kind}/${amount.toString()}`;
+}
+
+export const usdValue = async function (
+  token: Pick<TokenDetails, "symbol" | "address">,
+  amount: BigNumber,
+  network: string,
+): Promise<BigNumber> {
+  if (!(network in REFERENCE_TOKEN)) {
+    throw new Error("Unsupported network for computing USD value");
+  }
+  try {
+    const response = await axios.get(
+      apiPriceUrl({
+        baseToken: token.address,
+        quoteToken: REFERENCE_TOKEN[network].address,
+        kind: OrderKind.SELL,
+        amount,
+        network,
+      }),
+    );
+    return BigNumber.from(response.data.amount);
```

Comment:
nanonit: maybe just a tiny assert that `response.data.token` is quote token (price estimate units are hard)